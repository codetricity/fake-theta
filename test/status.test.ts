import {
  request,
  toMatchInvalidHeaderParameterError,
  toMatchInvalidParameterValueError,
} from './helper';

describe('POST /osc/commands/status', () => {
  it('should respond with results', async () => {
    const res = await request()
      .post('/osc/commands/status')
      .send({
        id: '10',
      })
      .expect(200);
    expect(res.body).toMatchSnapshot({
      results: {
        fileUrl: expect.stringMatching(/^(http|https):\/\/.+\/.+\/.+\/.+\.JPG/),
      },
    });
  });

  it('should respond invalidParameterValue', async () => {
    const res = await request()
      .post('/osc/commands/status')
      .send({
        fileUri: 'dummy',
      })
      .expect(400);
    toMatchInvalidParameterValueError(res);
  });
});

describe('THETA X POST /osc/commands/status', () => {
  it('should respond with results', async () => {
    const res = await request()
      .post('/osc/commands/status')
      .send({
        id: '10',
      })
      .set('emulating-theta-model', 'x')
      .expect(200);
    expect(res.body).toMatchSnapshot({
      results: {
        fileUrl: expect.stringMatching(/^(http|https):\/\/.+\/.+\/.+\/.+\.JPG/),
      },
    });
  });

  it('should respond invalidParameterValue', async () => {
    const res = await request()
      .post('/osc/commands/status')
      .send({
        fileUri: 'dummy',
      })
      .set('emulating-theta-model', 'x')
      .expect(400);
    toMatchInvalidParameterValueError(res);
  });
});

describe('THETA Z1 POST /osc/commands/status', () => {
  it('should respond with results', async () => {
    const res = await request()
      .post('/osc/commands/status')
      .send({
        id: '10',
      })
      .set('emulating-theta-model', 'z1')
      .expect(200);
    expect(res.body).toMatchSnapshot({
      results: {
        fileUrl: expect.stringMatching(
          /^(http|https):\/\/.+\/.+\/.+\/.+\/.+\.JPG/,
        ),
      },
    });
  });

  it('should respond invalidParameterValue', async () => {
    const res = await request()
      .post('/osc/commands/status')
      .send({
        fileUri: 'dummy',
      })
      .set('emulating-theta-model', 'z1')
      .expect(400);
    toMatchInvalidParameterValueError(res);
  });
});

describe('Unknown Type POST /osc/commands/status', () => {
  it('should respond invalidHeaderParameterError', async () => {
    const res = await request()
      .post('/osc/commands/status')
      .send({
        id: '10',
      })
      .set('emulating-theta-model', 'unknown')
      .expect(400);
    toMatchInvalidHeaderParameterError(res);
  });
});
