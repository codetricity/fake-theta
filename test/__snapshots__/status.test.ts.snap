// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`POST /osc/commands/status should respond with results 1`] = `
{
  "name": "camera.takePicture",
  "results": {
    "fileUrl": StringMatching /\\^\\(http\\|https\\):\\\\/\\\\/\\.\\+\\\\/\\.\\+\\\\/\\.\\+\\\\/\\.\\+\\\\\\.JPG/,
  },
  "state": "done",
}
`;

exports[`THETA X POST /osc/commands/status should respond with results 1`] = `
{
  "name": "camera.takePicture",
  "results": {
    "fileUrl": StringMatching /\\^\\(http\\|https\\):\\\\/\\\\/\\.\\+\\\\/\\.\\+\\\\/\\.\\+\\\\/\\.\\+\\\\\\.JPG/,
  },
  "state": "done",
}
`;

exports[`THETA Z1 POST /osc/commands/status should respond with results 1`] = `
{
  "name": "camera.takePicture",
  "results": {
    "fileUrl": StringMatching /\\^\\(http\\|https\\):\\\\/\\\\/\\.\\+\\\\/\\.\\+\\\\/\\.\\+\\\\/\\.\\+\\\\/\\.\\+\\\\\\.JPG/,
  },
  "state": "done",
}
`;
